src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/interceptor/InterceptorRule.java

```diff
@@ -0,0 +1,219 @@
+	/**
+	 *
+	 */
+	public InterceptorRule() {
+	}
+
+	/**
+	 *
+	 * @param rule
+	 * @param service
+	 * @param operation
+	 * @param endPoint
+	 */
+	public InterceptorRule(String event, String rule, String service,
+			String operation, String endPoint) {
+		this.event = event;
+		this.rule = rule;
+		this.service = service;
+		this.operation = operation;
+		this.endPoint = endPoint;
+	}
+
+	/**
+	 * @return the event
+	 */
+	public String getEvent() {
+		return event;
+	}
+
+	/**
+	 * @param event the event to set
+	 */
+	public void setEvent(String event) {
+		this.event = event;
+	}
+
+	/**
+	 * @return the rule
+	 */
+	public String getRule() {
+		return rule;
+	}
+
+	/**
+	 * @param rule the rule to set
+	 */
+	public void setRule(String rule) {
+		this.rule = rule;
+	}
+
+	/**
+	 * @return the service
+	 */
+	public String getService() {
+		return service;
+	}
+
+	/**
+	 * @param service the service to set
+	 */
+	public void setService(String service) {
+		this.service = service;
+	}
+
+	/**
+	 * @return the operation
+	 */
+	public String getOperation() {
+		return operation;
+	}
+
+	/**
+	 * @param operation the operation to set
+	 */
+	public void setOperation(String operation) {
+		this.operation = operation;
+	}
+
+	/**
+	 * @return the endPoint
+	 */
+	public String getEndPoint() {
+		return endPoint;
+	}
+
+	/**
+	 * @param endPoint the endPoint to set
+	 */
+	public void setEndPoint(String endPoint) {
+		this.endPoint = endPoint;
+	}
+
+	/**
+	 * @see java.lang.Object#hashCode()
+	 *
+	 *      According to
+	 *      http://stackoverflow.com/questions/27581/overriding-equals
+	 *      -and-hashcode-in-java it is recommended only to use getter-methods
+	 *      when using ORM like Hibernate
+	 */
+	@Override
+	public int hashCode() {
+		// two randomly chosen prime numbers
+		return new HashCodeBuilder(19, 89).
+				appendSuper(super.hashCode()).
+				append(getEvent()).
+				append(getRule()).
+				append(getService()).
+				append(getOperation()).
+				append(getEndPoint()).
+				toHashCode();
+	}
+
+	/**
+	 * @see java.lang.Object#equals(java.lang.Object)
+	 *
+	 *      According to
+	 *      http://stackoverflow.com/questions/27581/overriding-equals
+	 *      -and-hashcode-in-java it is recommended only to use getter-methods
+	 *      when using ORM like Hibernate
+	 */
+	@Override
+	public boolean equals(Object obj) {
+		if (!(obj instanceof InterceptorRule))
+			return false;
+		InterceptorRule other = (InterceptorRule) obj;
+
+		return new EqualsBuilder().
+				appendSuper(super.equals(other)).
+				append(getEvent(), other.getEvent()).
+				append(getRule(), other.getRule()).
+				append(getService(), other.getService()).
+				append(getOperation(), other.getOperation()).
+				append(getEndPoint(), other.getEndPoint()).
+				isEquals();
+	}
+
+	/* (non-Javadoc)
+	 * @see java.lang.Object#toString()
+	 */
+	@Override
+	public String toString() {
+		return ToStringBuilder.reflectionToString(this, ToStringStyle.DEFAULT_STYLE);
```

Comment:
I'd prefer to not use this reflection things. They have introduced issues elsewhere.